benchmark/assert/deepequal-buffer.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+const common = require('../common.js');
+const assert = require('assert');
+const bench = common.createBenchmark(main, {
+  n: [1e3],
+  len: [1e2],
+  strict: ['strict', 'nonstrict']
+});
+
+function main(conf) {
+  const n = +conf.n;
+  const len = +conf.len;
+  var i;
+
+  const data = Buffer.allocUnsafe(len);
+  const actual = Buffer.alloc(len);
+  const expected = Buffer.alloc(len);
+  data.copy(actual);
+  data.copy(expected);
+
+  if (conf.strict === 'strict') {
```

Comment:
In other benchmarks we've used the pattern:

```js
switch (conf.strict) {
  case 'strict':
    /** do stuff **/
  case 'nonstrict':
    /** do stuff **/
  default:
    throw new Error('Unsupported method');
}
```

This makes it easier to plug in new choices later and makes a code a bit easier to follow